packages/react/dismissable-layer/src/DismissableLayer.tsx

```diff
@@ -269,19 +273,19 @@ function useFocusOutside(onFocusOutside?: (event: FocusOutsideEvent) => void) {
 }
 
 function dispatchUpdate() {
-  const event = new Event(CONTEXT_UPDATE);
+  const event = new CustomEvent(CONTEXT_UPDATE);
```

Comment:
> I’ve changed every usage of the the Event constructor to be CustomEvent when a custom type is provided, this is a style change as either constructor is susceptible (it’s important to remember priority is inferred by the type passed, not which constructor used) the hope being that staying consistent will make deciding when to use dispatchDiscreteCustomEvent easier.

Interested for feedback on this, my understand is that these are equivalent aside from the optional `detail`, so I figured using `CustomEvent` for anything custom would make reasoning easier.